.gitpod.yml

```diff
@@ -1,35 +1,56 @@
+    # Instead of rebuilding the nginx-base frontend when environment variables change
+    # We build the frontend directly through 'npm run dev' to use the latest .env file
+    # All services are listed here under src/api/readme.md'
     before: |
-      eval $(gp env -e GITPOD_API_URL=$(gp url 3000))
-      eval $(gp env -e GITPOD_WEB_URL=$(gp url 8000))
-      eval $(gp env -e GITPOD_MICROSERVICE_URL=$(gp url 8443))
+      eval $(gp env -e API_URL=$(gp url 3000))
+      eval $(gp env -e WEB_URL=$(gp url 8000))
+      eval $(gp env -e API_HOST=$(gp url 8443))
+      eval $(gp env -e SERVICES="status image auth search posts feed-discovery users parser planet redis elasticsearch login firebase")
+
+      sed -r \
+          -e  "s@(.+=)http://localhost:8000(/[^ ]*)*@\1$WEB_URL\2@g" \
+          -e  "s@(.+=)http://localhost:3000(/[^ ]*)*@\1$API_URL\2@g" \
+          -e  "s@(.+=)http://localhost([^:]*)@\1$API_HOST\2@g" \
+          -e  "s@development\.yml@gitpod\.yml@" \
+          config/env.development > .env
     init: |
       npm install
-      docker-compose --env-file config/env.gitpod build
-    # GitPod workspace urls are not stable, we have to look up urls and build again
+      docker-compose --env-file .env build $SERVICES
     command: |
-      docker-compose --env-file config/env.gitpod up -d
+      docker-compose --env-file .env up -d $SERVICES
       npm run start
 
+  # Start the dev server after all services are running
+  - name: Terminal
+    openMode: split-right
+    command: |
+      gp await-port 3000 &&  npm run dev
+
   - name: Terminal
     openMode: tab-after
+    command: |
+      echo -e "\n\n\033[94m ======================================================"  && echo -e "\n\033[33m💪 Services are built and run automatically in the first terminal, the second one is free" && echo -e "\033[33m💡 If you are only developing on the frontend:" && echo -e "\033[33m    1. Switch to the first terminal and terminate the prebuilds: \e[91mCTRL/CMD + c" && echo -e "\033[33m    2. Install dependencies: $ \033[92mnpm run install:next" && echo -e "\033[33m    3. Copy staging env: $ \033[92mcp config/env.staging > .env" && echo -e "\033[33m    4. Start the dev server: $ \033[92mnpm run dev" && echo -e "\033[33m💡 If you need to test the backend:" && echo -e "\033[33m    1. Go grab a coffee and wait, the build is only done once for each commit" && echo -e "\n\033[94m ======================================================\n\n"
 
+# Exposing service ports
+# Overriding the default notification popup when a port is open
 ports:
   # NextJS frontend
   - port: 8000
     visibility: public
     onOpen: open-preview
-  # Microservice
-  - port: 8443
-    visibility: public
-    onOpen: ignore
   # Web API
   - port: 3000
     visibility: public
     onOpen: ignore
+  # Microservice server
+  - port: 8443
+    visibility: public
+    onOpen: ignore
   #  Redis server
   - port: 6379
```

Comment:
Since the front-end doesn't need to access redis or elasticsearch, do we need these ports open?

Reply:
There's also the port used in the `nginx` instance for e2e testing, I don't think we need to expose that one.
Also, there's a port 4000 expose, what's that for?

Reply:
Same with the ports for Firebase, we don't need to expose those I think?

Reply:
No, I agree. We shouldn't expose the internal tooling (redis, elastic, firebase) whenever we can avoid it.

Reply:
Agree, I'll keep them there and change their visibilities to `private` and set `onOpen: ignore` so we don't have a bunch of popups when these ports are open.